wgpu-hal/src/dx12/mod.rs

```diff
@@ -1558,3 +1559,114 @@ pub enum ShaderModuleSource {
     DxilPassthrough(DxilPassthroughShader),
     HlslPassthrough(HlslPassthroughShader),
 }
+
+#[repr(C)]
+#[derive(Debug)]
+struct MeshShaderPipelineStateStream {
+    root_signature: *mut Direct3D12::ID3D12RootSignature,
+    task_shader: Direct3D12::D3D12_SHADER_BYTECODE,
+    mesh_shader: Direct3D12::D3D12_SHADER_BYTECODE,
+    pixel_shader: Direct3D12::D3D12_SHADER_BYTECODE,
+    blend_state: Direct3D12::D3D12_BLEND_DESC,
+    sample_mask: u32,
+    rasterizer_state: Direct3D12::D3D12_RASTERIZER_DESC,
+    depth_stencil_state: Direct3D12::D3D12_DEPTH_STENCIL_DESC,
+    primitive_topology_type: Direct3D12::D3D12_PRIMITIVE_TOPOLOGY_TYPE,
+    rtv_formats: Direct3D12::D3D12_RT_FORMAT_ARRAY,
+    dsv_format: Dxgi::Common::DXGI_FORMAT,
+    sample_desc: Dxgi::Common::DXGI_SAMPLE_DESC,
+    node_mask: u32,
+    cached_pso: Direct3D12::D3D12_CACHED_PIPELINE_STATE,
+    flags: Direct3D12::D3D12_PIPELINE_STATE_FLAGS,
+}
+impl MeshShaderPipelineStateStream {
+    /// # Safety
+    /// Self must outlive the bytes (I think)
+    pub unsafe fn to_bytes(&self) -> Vec<u8> {
+        use Direct3D12::*;
+        let mut bytes = Vec::new();
+
+        macro_rules! push_subobject {
+            ($subobject_type:expr, $data:expr) => {{
+                // Ensure 8-byte alignment for the subobject start
+                let alignment = 8;
+                let padding = (alignment - (bytes.len() % alignment)) % alignment;
+                bytes.extend(core::iter::repeat(0).take(padding));
```

Comment:
```suggestion
                // Ensure 8-byte alignment for the subobject start
                let alignment = 8;
                let aligned_length = bytes.len().next_multiple_of(alignment);
                bytes.resize(aligned_length, 0);
```
More clear, and uses the more optimized `resize`